doc/windows_compute/Offloading_IP_checksums_and_other_tasks.md

```diff
@@ -0,0 +1,170 @@
+# Offloading IP Checksums and other tasks
+
+## Introduction
+
+> Note: Offloading basically means "let somebody else do it".
+
+There are [various tasks][ms-task-offload] related to the IP stack that can be offloaded,
+such as computing checksums or segmenting large TCP packets.
+
+It is beneficial to perform these offloads as late as possible. In the best case,
+the offloading is performed in the network card hardware, so eg. no CPU cycles
+are wasted on computing the TCP checksum.
+
+Sometimes the offloading is not supported by the hardware, but still can be performed in the software.
+This is best described on an example — the LSO (Large (TCP) Send Offload) means
+that for the most time spent in the kernel the TCP packet can be stored as one single [`NET_BUFFER`][nbl],
+bigger than a single Ethernet frame. That means that all operations like filtering, routing, etc.
+can operate efficiently (because they look at a single packet instead of many)
+and the actual segmenting will be performed just before passing the buffer
+to the hardware. In the case of container-to-container connection, the actual segmentation
+can even never happen! As they say, the fastest code is the code that never runs.
+
+## Offloading in Hyper-V Virtual Switches
+
+> Note: Refer to [the blog post about VMSwitches and offloading][vmswitch-offloads] for more information.
+
+Usually, the OS Networking stack asks the network device about types
+of offloading it supports and offloads only the tasks that the hardware
+is capable of offloading. With VMSwitch it's not that simple, because
+it's hard to propagate the hardware capabilities to all the ports
+in VMSwitch (especially because when sending a packet into VMSwitch,
+we don't know where will it end up: a container, a virtual machine,
+a real network card?).
+
+To sidestep this uneven-offload-support issue, VMSwitch **always**
+advertises on all its ports that most offloads are supported.
+Therefore, the vRouter must be prepared to handle NBLs
+with these offloads requested.
+
+The offloads that are always supported are:
+
+*   IP, TCP, UDP send checksum offload.
+*   LSO (Large Send Offload), which is basically TCP Segmentation offload.
+*   Receive checksum offload (VMSwitch with the help of hardware or by
+    itself will set appropriate flags that mark the validity of checksums
+    in _incoming_ packets).
+
+So yes, we must handle all these offload requests.
```

Comment:
> So yes

This clarifies that the following information is rephrased information that was already given before. Removing this could make the reader wonder whether the two pieces of information are the separate things or not.

> Actually, don't use we at all. Just say that the extension must handle all these offload requests.

Tried it:
> So yes, The VMSwitch Extension must handle all these offload requests.
But this relationship is symmetric — when sending packets
it can rely on support for these offloads on all vPorts!

I think that the benefit of using "we" in the second part of the sentence is that programmer reading this think will be more likely to remember "I can use this feature!". With "it", the programmer needs to consciously process that they're writing the code of the Extension, thus they can use it in their code.